### PR TITLE
Fixed SQL apostrophe (') incompatibility

### LINKED ISSUE #34442: The duplication of products doesn't allow apostrophes (') in the customization field
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

When you try to duplicate a product with apostrophes ('), a bug is triggered (FAILED_DUPLICATE_CUSTOMIZATION_FIELDS) causing the duplicated product to be incomplete, and impossible to access.

The error is:

PrestaShop\PrestaShop\Core\Domain\Product\Exception\CannotDuplicateProductException code 90: Cannot bulk insert into table customization_field_lang failed

### Expected behavior

The duplication should work

### Steps to reproduce

1. Login to the Back-office
2. Open the Products page
3. Try to duplicate a product with a customization field with an apostrophe (') such as "Merci d'insérer votre personnalisation"
4. An error appear

### PrestaShop version(s) where the bug happened

8.1.1

### PHP version(s) where the bug happened

8.2.7

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by Lunyyx**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | 8.1.x
| Description?      | Fixed the problem that was caused by apostrophes in SQL queries, this problem happened when the user write something like "Merci d'insérer votre personnalisation" as the apostrophe (') was considered as the end of the string.
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Try to duplicate an item with a apostrophe (') in the customization field
| Fixed issue or discussion?     | Fixes #34442

### CODE REVIEW

**Comment by Lunyyx** [comment]
> Hi, thanks for this contribution!
> 
> I found some issues with the Pull Request description:
> 
> * Your pull request does not seem to fix any issue, consider [creating one](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (see note below) and linking it by writing `Fixes #1234`.
> 
> Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!
> 
> ### About linked issues
> Please consider opening an issue before submitting a Pull Request:
> 
> * If it's a bug fix, it helps maintainers verify that the bug is effectively due to a defect in the code, and that it hasn't been fixed already.
> * It can help trigger a discussion about the best implementation path before a single line of code is written.
> * It may lead the Core Product team to mark that issue as a priority, further attracting the maintainers' attention.
> 
> (Note: this is an automated message, but answering it will reach a real human)

Fixed the old pull request

File: src/Adapter/Product/Update/ProductDuplicator.php [hunk 1]
```diff
@@ -1025,7 +1025,7 @@ private function bulkInsert(string $table, array $multipleRowValues, int $errorC
                     return 'null';
                 }
 
-                if(gettype($columnValue) == "string") {
+                if (gettype($columnValue) == 'string') {
```
Comment (Hlavtox) [inline src/Adapter/Product/Update/ProductDuplicator.php]:
```suggestion
                if (is_string($columnValue)) {
```

**Review by Hlavtox — APPROVED** [review]
I think it's OK. :-)

Some prepared statement or built in escape method would probably be better, but given the scope of that method and how it manually builds the query by joining the fields, it should work just fine. There should be no other possible fails that it could encounter, the rest of the query is not dynamic.

**Comment by M0rgan01** [comment]
Tests ui: https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/6731056352

**Comment by Hlavtox** [comment]
@M0rgan01 I think you ran the tests on develop instead of 8.1.x ;-)

**Comment by Lunyyx** [comment]
> I think it's OK. :-)
> 
> Some prepared statement or built in escape method would probably be better, but given the scope of that method and how it manually builds the query by joining the fields, it should work just fine. There should be no other possible fails that it could encounter, the rest of the query is not dynamic.

Perfect ! :)

Yes, I think it's weird that there isn't any buildt in functions to do this, but anyway, it should work without any problem yes

**Review by M0rgan01 — APPROVED** [review]

**Review by florine2623 — APPROVED** [review]
Hello @Lunyyx ,

The field is not duplicated anymore. The process to checkout is working as expected.

It is QA ✅ 
Thanks!

**Comment by Hlavtox** [comment]
@florine2623 I'm not sure I understand the message. :-)

This issue was about an error when duplicating a product in BO.
It didn't work before, now it should work with this PR.

**Review by Hlavtox — CHANGES_REQUESTED** [review]
Block to avoid accidental merge

**Comment by Lunyyx** [comment]
> @florine2623 I'm not sure I understand the message. :-)
> 
> This issue was about an error when duplicating a product in BO. It didn't work before, now it should work with this PR.

Yes it is, maybe I didn't explained correctly, it's just about duplication of products in the BO as you said.
Do I need to do something, or I just need to wait for someone to accept it ?

**Review by matthieu-rolland — APPROVED** [review]

**Comment by matthieu-rolland** [comment]
why shouldn't the PR be merged @Hlavtox  ?

**Comment by florine2623** [comment]
Hello @Hlavtox ,

The field was duplicated here in FO :
Screenshot 2023-11-02 at 15 17 07

And with the PR, the problem does not show up anymore ^^

Screenshot 2023-11-02 at 15 18 51

For me it is OK to be merged

**Comment by Hlavtox** [comment]
@florine2623 OK, but the problem was in the backoffice, as you can see on video from @paulnoelcholot here - https://github.com/PrestaShop/PrestaShop/issues/34442#issuecomment-1790449698

Snímek obrazovky 2023-11-02 152831

**Comment by florine2623** [comment]
Hello again @Hlavtox and @Lunyyx ,

Indeed, well it seems that 2 problems are fixed then !
No duplication in FO and no error while duplicating a product in BO, when using an apostrophe ✅ 

Although, now, I have a deprecation alert in FO whenever I duplicate a product. Whether there's an apostrophe or not :
`Deprecated: preg_match(): Passing null to parameter #2 ($subject) of type string is deprecated in /my_shop/classes/Validate.php on line 683`


Screenshot 2023-11-02 at 15 36 23

Could you check ?

**Comment by kpodemski** [comment]
@florine2623 is it really because of this PR? did you check it on installation without it?

**Comment by florine2623** [comment]
Hi again ^^

This PR can be merged as seen with @Hlavtox .
The other problem that I have is related to https://github.com/PrestaShop/PrestaShop/issues/33492 and can be fixed later on !

**Comment by Hlavtox** [comment]
We checked with florine privately and the issue is not related. So all good. :) Thanks everyone!

**Review by Hlavtox — APPROVED** [review]

**Pull request merged**